fetchcode/package_registry/__init__.py

```diff
@@ -0,0 +1,103 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import requests
+
+from packageurl import PackageURL
+from packageurl.contrib.route import Router
+from packageurl.contrib.route import NoRouteAvailable
+
+
+router = Router()
+
+
+class Data:
```

Comment:
Can you consider using `attr` or a dataclass for this? Ask @sbs2001 if you need help too.

Reply:
I agree. @TG1999 see: https://docs.python.org/3/library/dataclasses.html

Reply:
Btw if you are using dataclasses as @majurg  mentioned, you do need to consider that python3.7+ should be used.

Reply:
@TG1999 fetchcode should be using python3 already, no?

Reply:
Cool @majurg , pushing the code for same :)

Reply:
And what's the approach you will use then?

Reply:
We will be using class attributes :)

Reply:
You cannot use class attributes.... that does not work. You can use descriptors with attrs.org or dataclasses  but definitely not class attributes. A class __is__ a global... meaning its attributes are global too. There is only one class!
Here is the issue:
```
>>> class Foo:
...   name=None
...   version=None
... 
>>> def get_foo(name, version=None):
...   if name:
...     Foo.name = name
...   if version:
...     Foo.version = version
...   return Foo
... 
>>> a = get_foo(name='bar', version=12)
>>> a.name, a.version
('bar', 12)
>>> b = get_foo(name='BAZ')
>>> b.name, b.version
('BAZ', 12)
>>> a is b
True
>>> a==b
True
```

Reply:
Gotcha!!! Making changes for it.